lib/model/repositories/ratingEngine/implementation/MockRatingEngineRepository.dart

```diff
@@ -1,31 +1,22 @@
-import 'package:farmsmart_flutter/model/model/FactorEntity.dart';
 import 'package:farmsmart_flutter/model/model/mock/MockRecommendation.dart';
-import 'package:farmsmart_flutter/model/repositories/MockListRepository.dart';
 
 import '../RatingEngineRepositoryInterface.dart';
 
-final _mockWeights = harryWeights;
-
-class MockRatingEngineRepository extends MockListRepository<FactorEntity>
-    implements RatingEngineRepositoryInterface {
-  MockRatingEngineRepository._(
-      IdentifyEntity<FactorEntity> identifyEntity, List<FactorEntity> startData)
-      : super(
-          identifyEntity: identifyEntity,
-          startingData: startData,
-        );
-
-  factory MockRatingEngineRepository() {
-    return MockRatingEngineRepository._(null, _mockWeights);
-  }
-
+class MockRatingEngineRepository implements RatingEngineRepositoryInterface {
   @override
-  Future<List<FactorEntity>> getWeights() {
-    return getList();
+  Future<Map<String, RatingInfo>> getRatingInfo() {
+    final inputFactors = harryInputFactors.map((subject,value){
+      return MapEntry(subject, RatingInfo(harryWeights[subject], value));
+    });
+    return Future.value(inputFactors);
   }
 
   @override
-  Stream<List<FactorEntity>> observeWeights() {
-    return observeList();
+  Future<Map<String, RatingInfo>> observeRatingInfo() {
+    // TODO: implement observeRatingInfo
```

Comment:
Just to be sure, that we didn't miss it, that would go with another PR?

Reply:
In the mock it will never change, but I should implement to be complete. 👍

Reply:
done